src/server/ServerHandler.java

```diff
@@ -197,7 +195,39 @@ private GamingResponse handleRequestMove() {
      */
     @Override
     public void run() {
-        // Empty for now - will process client requests later
+        while (true) {
+            try {
+                // Read serialized request from client
+                String serializedRequest = dataInputStream.readUTF();
+                LOGGER.info("Received request: {}", serializedRequest);
+
+                // Deserialize request
+                Request request = gson.fromJson(serializedRequest, Request.class);
+
+                // Handle request and get response
+                Response response = handleRequest(request);
+
+                // Serialize and send response
+                String serializedResponse = gson.toJson(response);
+                dataOutputStream.writeUTF(serializedResponse);
+                dataOutputStream.flush();
+                LOGGER.info("Sent response: {}", serializedResponse);
```

Comment:
Change this to debug logging as well

```suggestion
                LOGGER.debug("Sent response: {}", serializedResponse);
```